prusti-encoder/src/encoders/type/predicate.rs

```diff
@@ -691,7 +691,10 @@ impl<'vir, 'tcx> PredicateEncValues<'vir, 'tcx> {
             // `Ref` is only part of snapshots for mutable references.
             data.snap_data
                 .field_snaps_to_snap
-                .apply(self.vcx, &[inner_snap, self_ref])
+                .apply(self.vcx, &[inner_snap, 
+                    // TODO: Re-enable this once its purpose is understood
+                    // self_ref
```

Comment:
The idea was, as the comment explains, that the snapshot of a mutable reference consists of both a snapshot of the pointee at the time of creation (IIRC) and the actual reference `Ref`. The latter is somewhat analogous to a prophecy. @JonasAlaif would have more info.

Reply:
Yea I conceptually figured that but since it wasn't typing I needed to do more digging